hedera-node/hedera-app/src/xtest/java/contract/MiscViewsXTest.java

```diff
@@ -0,0 +1,233 @@
+                            .ownerId(ERC_USER_ID)
+                            .metadata(Bytes.wrap("https://example.com/721/" + sn))
+                            .build());
+        }
+        return nfts;
+    }
+
+    @Override
+    protected Map<EntityIDPair, TokenRelation> initialTokenRelationships() {
+        final var tokenRelationships = new HashMap<EntityIDPair, TokenRelation>();
+        tokenRelationships.put(
+                EntityIDPair.newBuilder()
+                        .tokenId(ERC20_TOKEN_ID)
+                        .accountId(ERC_USER_ID)
+                        .build(),
+                TokenRelation.newBuilder()
+                        .tokenId(ERC20_TOKEN_ID)
+                        .accountId(ERC_USER_ID)
+                        .balance(111L)
+                        .kycGranted(true)
+                        .build());
+        tokenRelationships.put(
+                EntityIDPair.newBuilder()
+                        .tokenId(ERC721_TOKEN_ID)
+                        .accountId(ERC_USER_ID)
+                        .build(),
+                TokenRelation.newBuilder()
+                        .tokenId(ERC721_TOKEN_ID)
+                        .accountId(ERC_USER_ID)
+                        .balance(3L)
+                        .kycGranted(true)
+                        .build());
+        return tokenRelationships;
+    }
+
+    @Override
+    protected Map<TokenID, Token> initialTokens() {
+        final var tokens = new HashMap<TokenID, Token>();
+        final var erc20Symbol = "SYM20";
+        final var erc20Name = "20 Coin";
+        final var erc20Memo = "20 Coin Memo";
+        final var erc20Decimals = 2;
+        final var erc20TotalSupply = 888L;
+        final var erc721Symbol = "SYM721";
+        final var erc721Name = "721 Unique Things";
+        final var erc721Memo = "721 Unique Things Memo";
+        tokens.put(
+                ERC20_TOKEN_ID,
+                Token.newBuilder()
+                        .tokenId(ERC20_TOKEN_ID)
+                        .memo(erc20Memo)
+                        .name(erc20Name)
+                        .symbol(erc20Symbol)
+                        .decimals(erc20Decimals)
+                        .totalSupply(erc20TotalSupply)
+                        .tokenType(TokenType.FUNGIBLE_COMMON)
+                        .build());
+        tokens.put(
+                ERC721_TOKEN_ID,
+                Token.newBuilder()
+                        .tokenId(ERC721_TOKEN_ID)
+                        .memo(erc721Memo)
+                        .name(erc721Name)
+                        .symbol(erc721Symbol)
+                        .tokenType(TokenType.NON_FUNGIBLE_UNIQUE)
+                        .build());
+        return tokens;
+    }
+
+    @Override
+    protected RunningHashes initialRunningHashes() {
+        return RunningHashes.newBuilder().nMinus3RunningHash(PRNG_SEED).build();
+    }
+
+    @Override
+    protected void assertExpectedStorage(
```

Comment:
As mentioned above, the prng call will not change the state.  Also the view functions should also not update state.

Reply:
I get that these functions won't _change_ the state so I'm asking if it was not worth it to _check_ that.  I don't have strong feelings about it.

Reply:
I see what you mean but it's a little difficult the way the base class is currently laid out as we would have to take a snapshot of the state before running the prng function.  Let me think on it.

Reply:
Oh, I thought these things were checking individual maps that were initialized as part of the test.  I'll have to reread the code.

Reply:
I'm torn on this. On one hand, verifying changes to state could be very useful in determining correct behavior. On the other hand, state is intended to be encapsulated by the API, so testing it could be like running tests on database content, which feels funny

Reply:
I get your point.  It could be argued that this is a _view_ function thus you should check that state _did_ not change.  But that's why I was asking, not sure of the value.